spring-cloud-gcp-bigquery/src/main/java/org/springframework/cloud/gcp/bigquery/integration/outbound/BigQueryFileMessageHandler.java

```diff
@@ -62,75 +65,97 @@
  */
 public class BigQueryFileMessageHandler extends AbstractReplyProducingMessageHandler {
 
+	private static final String HEADER_FORMAT = "headers[%s]";
+
 	private final BigQuery bigQuery;
 
 	private final TaskScheduler taskScheduler;
 
-	private final EvaluationContext evaluationContext;
+	private EvaluationContext evaluationContext;
 
-	private Expression datasetNameExpr;
+	private Expression datasetNameExpression;
 
-	private Expression tableNameExpr;
+	private Expression tableNameExpression;
 
-	private FormatOptions formatOptions;
+	private Expression formatOptionsExpression;
 
 	private boolean autoDetectSchema = true;
 
 	private WriteDisposition writeDisposition = WriteDisposition.WRITE_APPEND;
 
 	private Duration timeout = Duration.ofMinutes(5);
 
+	private Duration jobPollInterval = Duration.ofSeconds(2);
+
 	private boolean sync = false;
 
 	public BigQueryFileMessageHandler(BigQuery bigQuery, TaskScheduler taskScheduler) {
 		Assert.notNull(bigQuery, "BigQuery client must not be null.");
 		this.bigQuery = bigQuery;
 		this.taskScheduler = taskScheduler;
+
+		ExpressionParser expressionParser = new SpelExpressionParser();
+		this.datasetNameExpression = expressionParser.parseExpression(
+				String.format(HEADER_FORMAT, BigQuerySpringMessageHeaders.DATASET_NAME));
```

Comment:
As I said: this must be a `FunctionExpression`. No reason to go a parsing and SpEL processing way: we just can call a `Function<Message, String>` instead.